modules/nlu/src/backend/api.ts

```diff
@@ -204,6 +204,26 @@ export default async (bp: typeof sdk, state: NLUState) => {
     const { botId, id } = req.params
     const botEngine = state.nluByBot[botId].engine1 as ScopedEngine
     await botEngine.storage.deleteEntity(id)
+
+    const intents = await botEngine.storage.getIntents()
+    const awaitsList = intents.map(async intent => {
+      const entitySlotIndex = intent.slots.findIndex(slot => slot.entities.includes(id))
+      if (entitySlotIndex === -1) {
+        return
+      }
+
+      const currentSlot = intent.slots[entitySlotIndex]
+      currentSlot.entities.splice(currentSlot.entities.indexOf(id), 1)
+
+      if (!currentSlot.entities.length) {
+        intent.slots.splice(entitySlotIndex, 1)
+      }
```

Comment:
If a slot is removed, we should also call `removeSlotFromUtterances()` to update `intent.utterances` as done [here](https://github.com/botpress/botpress/blob/master/modules/nlu/src/views/lite/intentEditor/IntentEditor.tsx#L48)

Reply:
removeSlotFromUtterances added, ty.

As for that bug, can't reproduce.
When I remove entity, it gets removed from slot, if no entities left, slot could be removed, including utterances(after current update)

Reply:
I think slot shouldn't be removed, it should be assigned to system.any if no entities left in the slot definition

Reply:
intent.slots.splice(entitySlotIndex, 1)

It just become empty array.